src/gen/gen_ast_builder.ml

```diff
@@ -117,11 +225,11 @@ struct
               let prefix =
                 common_prefix (List.map cds ~f:(fun cd -> cd.pcd_name.txt))
               in
-              List.map cds ~f:(fun cd ->
-                  gen_combinator_for_constructor ~wrapper path ~prefix cd))
+              List.map cds
+                ~f:(gen_combinator_for_constructor ~wrapper path ~prefix td))
       | Ptype_record lds ->
           let prefix = prefix_of_record lds in
-          [ gen_combinator_for_record path ~prefix lds ]
+          [ gen_combinator_for_record path ~prefix td td.ptype_attributes lds ]
```

Comment:
We could just pass `td` here and let `gen_combinator_for_record` extract the attributes from it couldn't we?

Reply:
Yep